### DISCUSSION TITLE
How to use Suspense with SSR?

### DISCUSSION THREAD

**Discussion by karimh777** [nuxt/nuxt#16631] [Questions]
When trying to use `Suspense` component in a page, the loading of this page is stuck till the async dependencies are resolved.

Navigating using `NuxtLink`  suspense component works normally. 

This doesn't happen if the Suspense component is wrapped in `ClientOnly` component.

[Play Demo](https://stackblitz.com/edit/github-pnm3sv-xpz768)

**Comment by ShourovRoy** [comment]
Use this instead > pages > shop > index.vue

```
<template>
  <div>
    <div>Shop Here</div>
    <ClientOnly>
      <Suspense>
        <template #default><AsyncComp></AsyncComp></template>
        <template #fallback>Loading.....</template>
      </Suspense>
      <template #fallback> loading in client... </template>
    </ClientOnly>
    <NuxtLink to="/">Home</NuxtLink>
  </div>
</template>

<script setup>
import AsyncComp from '../../components/asyncComp.vue';
</script>
```

**Reply by Rusinas** [reply]
Is it possible to use `Suspense` without `ClientOnly`?
